### ISSUE TITLE
Custom centered message in the middle or at the start of a discussion

### ISSUE THREAD

**Issue by guillaumeboussion** [flyerhq/flutter_chat_ui#649]
Hey, thanks for this package !

In order to build our chat page, we'd like to add some custom messages related to user actions in the middle or at the start of the chat bubbles (as a custom message type). This would look this way :
<img src="https://github.com/user-attachments/assets/af269203-bf3b-41c1-8b6d-605c241a7b40" width="200"/>
![]()

Is there any way to achieve this with the current version, or will this have to wait the v2 ?

**Comment by demchenkoalex** [comment]
hey! you sure this is a custom message and not just a widget at the top of the list that scrolls together with messages?

**Comment by guillaumeboussion** [comment]
Thanks for your answer @demchenkoalex 

Maybe that, in this specific case, this is the very first widget at the top of the scrollable widget, but what if we have to insert a custom message in the middle of the chat (ie: to show a scheduled appointment at some point in the chat)

**Comment by guillaumeboussion** [comment]
@demchenkoalex hey, any chance to achieve something like this ?

**Comment by demchenkoalex** [comment]
hey, yes, for sure, it is just end of a quarter at my job so lots to do and I also moved to another country, just need time to settle to come back to this, sorry about that

**Comment by demchenkoalex** [comment]
<img width="912" alt="Screenshot 2024-12-15 at 23 14 35" src="https://github.com/user-attachments/assets/ee2e96f7-8cbe-4449-96a2-fdde4d43af0e" />

I was able to achieve this effect with following `builders`:

```dart
builders: Builders(
  textMessageBuilder: (context, message) => FlyerChatTextMessage(message: message),
  imageMessageBuilder: (context, message) => FlyerChatImageMessage(message: message),
  customMessageBuilder: (context, message) => const Center(child: Text('Appointment scheduled')),
  chatMessageBuilder: (context, message, animation, child) =>
    ChatMessage(
      message: message,
      animation: animation,
      scaleAnimationAlignment: message is CustomMessage ? Alignment.center : null,
      alignment: message is CustomMessage ? Alignment.center : null,
      child: child,
    ),
),
```

hopefully this was the effect you were looking for?

**Comment by guillaumeboussion** [comment]
@demchenkoalex this looks perfect! Is this available without v2 ? Thanks to you man

**Comment by demchenkoalex** [comment]
without v2 you also use custom message and there is customMessageBuilder or something, I don't remember now, you will have to check v1 branch :)

**Comment by demchenkoalex** [comment]
released in [v2.0.0-dev.2](https://github.com/flyerhq/flutter_chat_ui/releases/tag/2.0.0-dev.2)
